src/main/listeners.ts

```diff
@@ -58,13 +60,31 @@ export function setupListeners() {
   ipc_getDesktopAppPath.main!.handle(async () => {
     // get the path and check existence
     const desktopAppPath = path.join(app.getPath("appData"), "Slippi Desktop App");
-    const exists = await fs.pathExists(desktopAppPath);
+    let exists = await fs.pathExists(desktopAppPath);
 
+    if (!exists) {
+      return { dolphinPath: "", exists: false };
+    }
+
+    // Linux doesn't need to do anything because their dolphin settings are in a user config dir
     if (isLinux && exists) {
       await fs.remove(desktopAppPath);
-      return { exists: false };
+      return { dolphinPath: "", exists: false };
     }
 
-    return { exists: exists };
+    const dolphinFolderPath = path.join(desktopAppPath, "dolphin");
+    exists = await fs.pathExists(dolphinFolderPath);
+
+    const dolphinExecutablePath = await findDolphinExecutable(DolphinLaunchType.NETPLAY, dolphinFolderPath);
+
+    return { dolphinPath: dolphinExecutablePath, exists: exists };
+  });
```

Comment:
Since this endpoint is now really only in charge of fetching and returning the old dolphin path, do you think we should move this endpoint into the `@dolphin/ipc` module instead?

Reply:
yeah i wasn't sure about that part, also the name is kinda weird. It checks the existence of the desktop app folder and then the dolphin folder. I think i'll rename it and move it.